session/pingpong/settlement_history_storage.go

```diff
@@ -60,17 +61,81 @@ func (shs *SettlementHistoryStorage) Store(she SettlementHistoryEntry) error {
 	return shs.bolt.DB().From(settlementHistoryBucket).Save(&she)
 }
 
-// Get returns the settlement history for given provider accountant combination.
-func (shs *SettlementHistoryStorage) Get(provider identity.Identity, accountant common.Address) ([]SettlementHistoryEntry, error) {
-	query := shs.bolt.DB().
-		From(settlementHistoryBucket).
-		Select(
-			q.Eq("ProviderID", provider),
-			q.Eq("AccountantID", accountant),
-		).
+// Query executes given query.
+func (shs *SettlementHistoryStorage) Query(query *SettlementHistoryQuery) (err error) {
+	return query.run(shs.bolt.DB().From(settlementHistoryBucket))
+}
+
+// NewSettlementHistoryQuery creates instance of query.
+func NewSettlementHistoryQuery() *SettlementHistoryQuery {
+	return &SettlementHistoryQuery{}
+}
+
+// SettlementHistoryQuery defines all flags for filtering in settlement history storage.
+type SettlementHistoryQuery struct {
+	Entries []SettlementHistoryEntry
```

Comment:
What's the benefit of combining a query with the result?

You could simply have query as follows without repeating 'filter':
```go
type SettlementHistoryQuery struct {
	From         *time.Time
	To           *time.Time
	ProviderID   *identity.Identity
	AccountantID *common.Address
}
```

And return `Entries []SettlementHistoryEntry` as a func return value.

Reply:
Simplified to 
```
type settlementHistoryProvider interface {
	List(filter pingpong.SettlementHistoryFilter) ([]pingpong.SettlementHistoryEntry, error)
}
```